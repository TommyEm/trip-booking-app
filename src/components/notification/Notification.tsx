import React from 'react';
import { Toast } from 'react-bootstrap';

import { StyledNotification } from './Notification.styled';


export interface INotificationProps {
	className?: string;
	isShown: boolean;
	tripInfo: string;
	onClose: () => void;
}

export const Notification: React.FC<INotificationProps> = ({
	className,
	isShown,
	tripInfo,
	onClose,
}: INotificationProps) => {
    return (
		<StyledNotification
			className={`Notification ${className}`}
			data-testid='test-Notification'
		>
			<Toast
				onClose={onClose}
				show={isShown}
				delay={3000}
				autohide
			>
				<Toast.Header>
					<strong className="me-auto">
						Your trip is booked!
					</strong>
				</Toast.Header>
				<Toast.Body>
					{tripInfo}
				</Toast.Body>
			</Toast>
		</StyledNotification>
	);
}
