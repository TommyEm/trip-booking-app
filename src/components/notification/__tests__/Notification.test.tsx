import React from 'react';
import {
	cleanup,
	render,
	screen
} from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from '../Notification.stories';

const { Default } = composeStories(stories);


afterEach(cleanup);
const storyComponent = <Default />;

it('should render a notification', () => {
	render(storyComponent);
	expect(screen.getByTestId('test-Notification')).toBeVisible();
});

it('matches snapshot', () => {
	const tree = render(storyComponent);
	expect(tree).toMatchSnapshot();
});
