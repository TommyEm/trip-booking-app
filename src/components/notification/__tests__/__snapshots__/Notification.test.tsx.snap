// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL gGYjnV Notification undefined"
        data-testid="test-Notification"
      >
        <div
          aria-atomic="true"
          aria-live="assertive"
          class="fade toast show"
          role="alert"
        >
          <div
            class="toast-header"
          >
            <strong
              class="me-auto"
            >
              Your trip is booked!
            </strong>
            <button
              aria-label="Close"
              class="btn-close"
              data-dismiss="toast"
              type="button"
            />
          </div>
          <div
            class="toast-body"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL gGYjnV Notification undefined"
      data-testid="test-Notification"
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        class="fade toast show"
        role="alert"
      >
        <div
          class="toast-header"
        >
          <strong
            class="me-auto"
          >
            Your trip is booked!
          </strong>
          <button
            aria-label="Close"
            class="btn-close"
            data-dismiss="toast"
            type="button"
          />
        </div>
        <div
          class="toast-body"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
